java-frontend/src/test/java/org/sonar/java/model/JMethodSymbolTest.java

```diff
@@ -154,6 +154,16 @@ void testUnknownInheritance() {
       retrieveMethodSymbol("KnownInterface", "a", cu));
   }
 
+  @Test
+  void testUnknownExtends() {
```

Comment:
To be fair, this test does not reproduce the problem reported in the ticket, I did not manage to reproduce the NPE...

Reply:
So the test is probably useless, and does not reflect what we want to try. I would not add it, to be honest.

In your test I don't see anything unknown as well. What about starting from a method invocation?
it's also maybe possible to manually craft the situation by mocking some incomplete ECJ binding.

Reply:
Indeed, the type extended was supposed to be Unknown. I think it makes sense to keep it (the corrected version), to test that it behaves correctly both with an unknown parent and unknown interfaces.

Testing this code with mock turns out to be a bit tricky: it is not possible to mock/spy `JMethodSymbol` (it is final), and not easy to mock the arguments either, because the constructor is using an external helper method. I ended up making the method visible for testing.